workflows/detaxizer.nf

```diff
@@ -105,95 +121,246 @@ workflow DETAXIZER {
     //
     // MODULE: Run fastp
     //
+    if (params.preprocessing) {
+
     FASTP (
         RENAME_FASTQ_HEADERS_PRE.out.fastq,
         [],
+        [],
         params.fastp_save_trimmed_fail,
         []
     )
+
+    ch_fastq_for_classification = FASTP.out.reads
     ch_versions = ch_versions.mix(FASTP.out.versions.first())
+    } else {
+        ch_fastq_for_classification = RENAME_FASTQ_HEADERS_PRE.out.fastq
+    }
+    //////////////////////////////////////////////////
+    //  Classification
+    //////////////////////////////////////////////////
```

Comment:
It seems like you have 3 blocks:
- kraken2 and bbduk
- bbduk only
- kraken2 only

and therefore there are repetitive code snippets. If I didnt misinterpret, then this maybe could be solved with less repetitive code using empty channels or channels that collect specific key outcome (or are empty if e.g. kraken2 is skipped)?